checkers/rules/rules.go

```diff
@@ -736,3 +736,24 @@ func dynamicFmtString(m dsl.Matcher) {
 		Suggest("errors.New($f($*args))").
 		Report(`use errors.New($f($*args)) or fmt.Errorf("%s", $f($*args)) instead`)
 }
+
+//doc:summary Detects strings.Compare usage
+//doc:tags    style experimental
+//doc:before  strings.Compare(x, y)
+//doc:after   x < y
+func stringsCompare(m dsl.Matcher) {
+	m.Match(`strings.Compare($s1, $s2) == 0`, `0 == strings.Compare($s1, $s2)`).
```

Comment:
There is no need to add a yoda-style pattern here.

They should be reported by a separate checker that just checks for `0 == $_` (not exactly, but that's the idea).

Reply:
\+